File: daily/C++版本/1290. 二进制链表转整数.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        // ListNode *p=head;
        // int flag=0;
        // while(p)
        // {
        //     flag++;
        //     p=p->next;
        // }
        // int sum=0;
        // while(head)
        // {
        //     sum+=head->val*pow(2,--flag);
        //     head=head->next;
        // }
        // return sum;
        int sum=0;
        while(head)
        {
            sum=sum*2+head->val;
            head=head->next;
        }
        return sum;
    }
};

File: daily/链表/链表easy/1290. 二进制链表转整数.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var getDecimalValue = function(head) {
    // var p=head,flag=0;
    // while(p)
    // {
    //     flag++;
    //     p=p.next;
    // }
    // var sum=0;
    // while(head)
    // {
    //     sum+=head.val*Math.pow(2,--flag);
    //     head=head.next;
    // }
    // return sum;

    var sum=0;
    while(head)
    {
        sum=sum*2+head.val;
        head=head.next;
    }
    return sum;
};